mbrl/util/env_handler.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+from typing import Optional, Tuple, Union, cast
+
+import gym
+import gym.wrappers
+import hydra
+import numpy as np
+import omegaconf
+import torch
+
+import mbrl.planning
+import mbrl.types
+
+from abc import ABC, abstractclassmethod, abstractmethod
+
+class Freeze(ABC):
+    """ Abstract base class for freezing various gym backends """
+    def __enter__(self, env):
+        raise NotImplementedError
+
+    def __exit__(self, env):
+        raise NotImplementedError
+
+def create_handler(env_name: str):
+    """Creates a new environment handler from its string description.
+
+    Args:
+        env_name (str): the string description of the environment. Valid options are:
+
+          - "dmcontrol___<domain>--<task>": a Deep-Mind Control suite environment
+            with the indicated domain and task (e.g., "dmcontrol___cheetah--run".
+          - "gym___<env_name>": a Gym environment (e.g., "gym___HalfCheetah-v2").
+          - "cartpole_continuous": a continuous version of gym's Cartpole environment.
+          - "pets_halfcheetah": the implementation of HalfCheetah used in Chua et al.,
+            PETS paper.
+          - "ant_truncated_obs": the implementation of Ant environment used in Janner et al.,
+            MBPO paper.
+          - "humanoid_truncated_obs": the implementation of Humanoid environment used in
+            Janner et al., MBPO paper.
+
+    Returns:
+        (EnvHandler): A handler for the associated gym environment
+    """
+    pass
+
+class EnvHandler(ABC):
+    """ Abstract base class for handling various gym backends
+
+    Subclasses of EnvHandler should define an associated Freeze subclass
+    and override self.freeze with that subclass """
+    
+    freeze: Freeze
+
+    @abstractclassmethod
+    def _legacy_make_env(
```

Comment:
We probably don't need `legacy_make_env`, since `v0.2.0`  adds a new mechanism to create environments based on Hydra, which is more scalable. See [this](https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/examples/conf/overrides/planet_cartpole_balance.yaml) example, and [here](https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/util/mujoco.py#L153) is how it's handled in the code, which can probably be added as the default `make_env` to the base `EnvHandler` class. 

Also, to make the function more user friendly, we can also allow a simple python dict to be passed, as it is done [here](https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py#L80). Incidentally, we don't have to add [this]( https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py#L94), since I checked that `hydra.utils.instantiate` works if you just pass a dict.

Reply:
Looks good! I will use `mbrl.util.mujoco.make_env` as the default `make_env` for `EnvHandler`. I'll also update the type hints to note `cfg` argument can also be a simple python dict without any changes to the underlying code.

Reply:
I found that in the end it's still necessary for `cfg` to be an `OmegaConf` because `make_env` directly accesses attributes of `cfg` e.g. `cfg.overrides`. I fixed this by applying (this)[https://github.com/facebookresearch/mbrl-lib/blob/main/mbrl/models/gaussian_mlp.py#L94]